runtime/lua/vim/provider/perl.lua

```diff
@@ -0,0 +1,66 @@
+local M = {}
+local s_err ---@type string?
+local s_host ---@type string?
+
+function M.require(host, prog)
+  local args = { prog, '-e', 'use Neovim::Ext; start_host();' }
+
+  -- Collect registered perl plugins into args
+  local perl_plugins = vim.fn['remote#host#PluginsForHost'](host.name) ---@type any
+  ---@param plugin any
+  for _, plugin in ipairs(perl_plugins) do
+    table.insert(args, plugin.path)
+  end
+
+  return vim.fn['provider#Poll'](args, host.orig_name, '$NVIM_PERL_LOG_FILE')
+end
+
+--- @return string? path to detected perl, if any; nil if not found
+--- @return string? error message if perl can't be detected; nil if success
+function M.detect()
+  -- use g:perl_host_prog if set or check if perl is on the path
+  local prog = vim.fn.exepath(vim.g.perl_host_prog or 'perl')
+  if prog == '' then
+    return nil, 'No perl executable found'
+  end
+
+  -- if perl is available, make sure we have 5.22+
+  vim.fn.system({ prog, '-e', 'use v5.22' })
```

Comment:
`vim.system` does not work here, I think there's a bug.

Reply:
interesting. what happens exactly?

Reply:
`vim.system` just silently aborts in one of the tests, causing it to fail.